tests/run-make/wasm-custom-sections-opt/rmake.rs

```diff
@@ -0,0 +1,30 @@
+extern crate run_make_support;
+
+use run_make_support::{out_dir, rustc, wasmparser};
+use std::collections::HashMap;
+use std::path::Path;
+
+fn main() {
+    if std::env::var("TARGET").unwrap() != "wasm32-wasip1" {
+        return;
+    }
+
+    rustc().arg("foo.rs").arg("--target=wasm32-wasip1").arg("-O").run();
```

Comment:
Do you mean something like an `output_file()` on the return value from `run()`? 

If so, that may be somewhat difficult as the precise output file name depends on the target being used, for example it'd have to parse `--target` passed here and realize that the output file extension is `*.wasm`.

Reply:
> Do you mean something like an `output_file()` on the return value from `run()`?

Sorry I meant for the input file, predicting the output file name is indeed a massive pain and shouldn't be done (just let the linker et al. do that) 😆

Reply:
Ah ok I see, but at least in my experience you rarely want to just take `&Path` when it's common to take string literals, which typically leads to something like `path: impl AsRef<Path>`.

Either that or put another way the boilerplate of requiring `Path::new("some literal")` probably won't necessarily make it too much harder to shoot yourself in the foot with paths (vs just passing `"some literal"`).

From above, though, could you elaborate on:

> because some tools/executables properly normalize/convert a Unix-y input path so they work on Windows properly, but other tools just give up

AFAIK native Windows APIs, which anything going through Rust uses (e.g. the `run_make_support` crate), will never normalize anything. Instead Windows allows both `/` and `\` as path separators which is why `c:\a/b\c` works. I believe that "verbatim paths" which IIRC are something like `\\?\...` (or something like that sorry I forget the specifics) require that the path separator is `\`, not `/`, but AFAIK that's pretty rare.

That's all to say, while I know that mingw/cygwin/etc can cause confusion with paths I've rarely run into `/` not working as a path separator except in the rare case of verbatim paths.

Reply:
> That's all to say, while I know that mingw/cygwin/etc can cause confusion with paths I've rarely run into `/` not working as a path separator except in the rare case of verbatim paths.

Yeah it's probably fine, I think I may have run into some strange path issues before.